doc/src/index-fileformats.md

```diff
@@ -0,0 +1,119 @@
+# Tantivy index file formats
+This document defines the index file formats used in Tantivy. 
+
+Tantivy has one metadata json file for each index and a few files for each segment, each file storing a specific datastructure.
+
+## Index metadata json
+For a given index, Tantivy stores the following metadata in a json file `meta.json` :
+- the list of segments with its metadata id, max_doc, deletes ;
+- the index schema represented by the list of fields with name, type and option data ;
+- the opstamp or timestamp of the operation (commit).
+
+```
+{
+  "segments": [
+    {
+      "segment_id": uuid string,
+      "max_doc": int|null,
+      "deletes": int|null
+    },
+    ...
+  ],
+  "schema": [
+    {
+      "name": string,
+      "type": string,
+      "options": {
+        "indexing": {
+          "record": string,
+          "tokenizer": string
+        },
+        "stored": bool
+      }
+    },
+ ],
+ ...
+ "opstamp": int
+}
+```
+
+
+## Endianness
+All integers and floats are serialiazed in litte indian order.
```

Comment:
That's actually incorrect. In terms, we use bigendian explicitly. That's because big endian lexicographical order is the natural order of integer.

Reply:
(and we use little endian everywhere else because modern CPUs are little endian)

Reply:
nice, I was pretty sure I was missing something but did not remind where bigendian was used

Reply:
+